### PR TITLE
Pass through versioning properties to the pack command

### PR DESCRIPTION
**Pull request by weshaggard**
PTAL @mitchdenny 

cc @Azure/azure-sdk-eng

### CODE REVIEW

**Comment by mitchdenny** [comment]
LGTM. Template parameters are injected at queue time. The way to think of templates is:

```
yamlstring = mytemplate(parameter1, parameter2)
```

Templates are processed first and then environment variables are read at the time of execution.

**Review by mitchdenny — APPROVED** [review]

File: eng/pipelines/client.yml [hunk 1]
```diff
@@ -13,8 +18,7 @@ pr:
 trigger:
 - master
 
-variables:
-  ProjectFile: eng/data-plane.proj
```
Comment (mitchdenny) [inline eng/pipelines/client.yml]: Any particular reason that you removed this variable?

**Review by mitchdenny — DISMISSED** [review]
We'll need to put the variable back in for the data-plane project. However we should consider moving to template parameters in a lot of these scenarios.

File: eng/pipelines/client.yml (hunk 1 above)
Comment (weshaggard) [reply eng/pipelines/client.yml]: We have to remove the variable otherwise we cannot set it at queue time.

**Comment by weshaggard** [comment]
@mitchdenny the auto-triggering of all the pipelines is not going to work. Some set of pipelines always fail with the github "abuse detection mechanism". When they fail in that manner there also doesn't seem to be any easy way to re-run just that one leg. The "re-run" option on the checks page doesn't seem to work.

This seems like it is going to be a big issue for our current strategy so we need to keep pushing to find a good solution or potentially change our strategy.

**Comment by mitchdenny** [comment]
> @mitchdenny the auto-triggering of all the pipelines is not going to work. Some set of pipelines always fail with the github "abuse detection mechanism". When they fail in that manner there also doesn't seem to be any easy way to re-run just that one leg. The "re-run" option on the checks page doesn't seem to work.
> 
> This seems like it is going to be a big issue for our current strategy so we need to keep pushing to find a good solution or potentially change our strategy.

Agreed. Working with the GH and Azure Pipelines folks to find a solution.

**Pull request merged**